src/scripts/compareExtensions.php

```diff
@@ -0,0 +1,196 @@
+  if ($a == 'name') {
+    $return = -1; // if we make it 'less' it will be first in the array
+  } else if ($b == 'name') {
+    $return = 1;
+  } else {
+    $return = 0;
+  }
+//  echo "Returning $return\n";
+  return $return;
+}
+
+$HELP = <<<HERE
+This script helps you identify different extensions used in a 3rd-party wiki compared to Meza. It does not deploy Meza. It is only an informational tool.
+
+It actually does a three-way comparison between those 3rd-party "custom" extensions to whatever you've setup in
+/opt/conf-meza/public/MezaLocalExtensions.yml as your "improved" Meza.  The intention is to ensure that at the end of the day, your MezaLocalExtensions file will accurately include any and all extensions which are found on the 3rd-party wiki, but not found in Meza core.
+
+ Note that if MezaLocalExtensions.yml is not yet setup, this script will use the example
+ provided at config/core/MezaLocalExtensions.example.yml
+ (which serves as an example for what your local Extensions specification could look like.)
+
+Supply the full URL to the public wiki API as the script argument. e.g.
+`php compareExtensions.php https://freephile.org/w/api.php`
+to get a list of extensions which would need to be added locally.
+
+Sometimes, 3rd party wikis are not publicly accessible. In that case, this script can use a .json file to represent the external wiki. If the 3rd-party wiki API is access restricted, then execute a
+'?action=query&meta=siteinfo&siprop=extensions&format=json' API query in that
+environment and save the text output to a file with a .json extension.
+
+Then supply the path to that file as the argument to the script.
+HERE;
+
+/////////////////////// Configuration ///////////////////////////////////////////////////////////////////
+
+$mezaCoreExtensionsFile = '/opt/meza/config/core/MezaCoreExtensions.yml';
+$mezaCoreExtensionsURL = 'https://raw.githubusercontent.com/freephile/meza/master/config/core/MezaCoreExtensions.yml';
+$localExtensionsFile = '/opt/conf-meza/public/MezaLocalExtensions.yml';
+
+$coreYml = ( file_exists( $mezaCoreExtensionsFile ) ) ? file_get_contents($mezaCoreExtensionsFile) : file_get_contents($mezaCoreExtensionsURL) ;
+$localYml = ( file_exists( $localExtensionsFile ) ) ? file_get_contents($localExtensionsFile) : file_get_contents('/opt/meza/config/core/MezaLocalExtensions.example.yml') ;
+
+# normalize names by removing any spaces
+# grep -Po 'name\: (.*)'  /opt/meza/config/core/MezaCoreExtensions.yml | sort | awk --field-separator=: '{print $2}' | sed s'/[ \t]*//g';
+
+$pattern = '#name\: (.*)#';
+preg_match_all ($pattern, $coreYml, $matches);
+$core = $matches[1];
+preg_match_all ($pattern, $localYml, $matches);
+$local = $matches[1];
+sort($core);
+$core = str_replace(' ', '', $core);
+sort($local);
+$local = str_replace(' ', '', $local);
+
+$api = $argv[1];
+// show help if no argument is supplied
+if ( ($api == null) || ($api == false) || ($api == '') ) {
+  fwrite(STDERR, $HELP);
```

Comment:
It was not clear to me from reading the help that I needed to supply anything as an argument. It should probably start with a "usage" statement. Examples:

```
$ sed --help
Usage: sed [OPTION]... {script-only-if-no-other-script} [input-file]...

...more details...

$ grep --help
Usage: grep [OPTION]... PATTERN [FILE]...
Search for PATTERN in each FILE or standard input.
PATTERN is, by default, a basic regular expression (BRE).
Example: grep -i 'hello world' menu.h main.c

...more details...

$ meza --help
Mediawiki EZ Admin

Usage: meza COMMAND [directives]

...more details...
```